Source/WebGPU/WGSL/AST/TypeDecl.h

```diff
@@ -35,6 +36,7 @@ class TypeDecl : public ASTNode {
 
 public:
     enum class Kind : uint8_t {
+        Array,
```

Comment:
Arrays are parameterized, are they not? Why not add to the existing `ParameterizedType` class?

Reply:
I've flip-flopped on this and I bit the bullet and tried to implement array are as `ParameterizedType` but with the possibility of an `Expression` or `TypeDecl` in the 2nd type position, both of which contain `Identifier`, I'm going to go with this code. (eg. `array<f32, foo>` is `foo` a `TypeDecl` or `Expression`?)

I'm happy to re-visit this decision down the road.